addons/2d-color-picker/paint-editor.js

```diff
@@ -0,0 +1,210 @@
+      addon.tab.redux.removeEventListener("statechanged", onEyeDropperOpened);
+      setTimeout(() => {
+        const previousTool = addon.tab.redux.state.scratchPaint.color.eyeDropper.previousTool;
+        if (previousTool) previousTool.activate();
+        addon.tab.redux.state.scratchPaint.color.eyeDropper.callback(hex);
+        addon.tab.redux.dispatch({
+          type: "scratch-paint/eye-dropper/DEACTIVATE_COLOR_PICKER",
+        });
+      }, 50);
+    };
+    addon.tab.redux.addEventListener("statechanged", onEyeDropperOpened);
+    element.children[1].children[0].click();
+  };
+
+  // for the color picker's background color
+  const convertToGeneralColor = (hex) => {
+    let h = tinycolor(hex).toHsv();
+    h.s = 1;
+    h.v = 1;
+    return tinycolor(h).toHex();
+  };
+
+  // le loop
+  while (true) {
+    // wait for color dialog box appearance
+    const element = await addon.tab.waitForElement('div[class*="color-picker_swatch-row"]', { markAsSeen: true });
+    rateLimiter.abort(false);
+
+    // update the bg color of the picker
+    function updateColor() {
+      rateLimiter.limit(() => {
+        let c = getColor(element);
+        let chsv = tinycolor(c).toHsv();
+        updateHandleFinal(chsv.s, chsv.v);
+        saColorPicker.style.background = "#" + convertToGeneralColor(getColor(element));
+      });
+    }
+
+    // redux stuff
+    addon.tab.redux.initialize();
+    addon.tab.redux.addEventListener("statechanged", (e) =>
+      e.detail.action.type == "scratch-paint/fill-style/CHANGE_FILL_COLOR" ||
+      e.detail.action.type == "scratch-paint/fill-style/CHANGE_FILL_COLOR_2" ||
+      e.detail.action.type == "scratch-paint/stroke-style/CHANGE_STROKE_COLOR" ||
+      e.detail.action.type == "scratch-paint/stroke-style/CHANGE_STROKE_COLOR_2"
+        ? updateColor()
+        : 0
+    );
+    if (addon.tab.redux && typeof prevEventHandler === "function") {
+      addon.tab.redux.removeEventListener("statechanged", prevEventHandler);
+      prevEventHandler = null;
+    }
+
+    // get the color
+    if (addon.tab.editorMode !== "editor") continue;
+    let defaultColor = getColor(element);
+
+    // create the color picker element and all it's child elements
+    const saColorPicker = document.createElement("div");
+    saColorPicker.className = "sa-2dcolor-picker";
+    saColorPicker.style.background = "#" + convertToGeneralColor(defaultColor || "ff0000");
+
+    const saColorPickerImage = Object.assign(document.createElement("img"), {
+      className: "sa-2dcolor-picker-image",
+      src: addon.self.dir + "/assets/sv-gr.png",
+      draggable: false,
+    });
+    const saColorPickerHandle = Object.assign(document.createElement("div"), {
+      className: "slider_handle_3f0xk",
+    });
+    saColorPickerHandle.style.pointerEvents = "none";
+
+    // create the label
+    const saColorLabel = document.createElement("div");
+    saColorLabel.className = "color-picker_row-header_173LQ sa-2dcolor-label";
```

Comment:
For this to be implemented in TW (and other mods) this would not work because of the last part (_XXXXX). Look into `addon.tab.scratchClass()`. I think some addons like variable-manager use this.

Reply:
fixed!